File: codes/julia/1.Loops.jl
# while loop

i = 1
while true
    global i
    if (i === 2)
        i += 1
        if (i > 5)
            break
        end
        continue
    end
    println("while loop v1, i is ", i)
    i += 1
    if (i > 5)
        break
    end
end

i = 1
while (i <= 5)
    global i
    if (i === 2)
        i += 1
        continue
    end
    println("while loop v2, i is ", i)
    i += 1
end

# for loop

for i in 1:1:5
    if (i === 2)
        continue
    end
    println("for loop, i is ", i)
end

File: codes/python/1.Loops.py
# while loop

i = 1
while True:
    if (i == 2):
        i += 1
        if (i > 5):
            break
        continue
    print(f"while loop v1, i is {i}")
    i += 1
    if (i > 5):
        break

i = 1
while (i <= 5):
    if (i == 2):
        i += 1
        continue
    print(f"while loop v2, i is {i}")
    i += 1

# for loop

for i in range(1, (5 + 1), 1): # (start, stop, step)
    if (i == 2):
        continue
    print(f"for loop ascending, i is {i}")

for i in range(5, (1 - 1), -1): # (start, stop, step)
    if (i == 2):
        continue
    print(f"for loop descending, i is {i}")
